SS_objfunc.tpl

```diff
@@ -96,69 +97,56 @@ FUNCTION void evaluate_the_objective_function()
           }
           else //  no observations
           {
             Q_parm(Q_setup_parms(f, 1)) = Svy_log_q(f, 1);
           }
-
-          if (Svy_errtype(f) == -1) // normal
-          {
-            Svy_q(f) = Svy_log_q(f); //  q already in  arithmetic space
-          }
-          else
-          {
-            Svy_q(f) = mfexp(Svy_log_q(f)); // get q in arithmetic space
-          }
         }
 
         // SS_Label_Info_25.1.4 #calc the logL
-        if (Svy_errtype(f) == 0) // lognormal
-        {
-          for (i = 1; i <= Svy_N_fleet(f); i++)
-            if (Svy_use(f, i) > 0)
-            {
-              surv_like(f) += 0.5 * square((Svy_obs_log(f, i) - Svy_est(f, i)) / Svy_se_use(f, i)) + sd_offset * log(Svy_se_use(f, i));
-              //            should add a term for 0.5*s^2 for bias adjustment so that parameter approach will be same as the  biasadjusted scaling approach
-            }
-        }
-        else if (Svy_errtype(f) > 0) // t-distribution
-        {
-          dvariable df = Svy_errtype(f);
-          for (i = 1; i <= Svy_N_fleet(f); i++)
-            if (Svy_use(f, i) > 0)
-            {
-              surv_like(f) += ((df + 1.) / 2.) * log((1. + square((Svy_obs_log(f, i) - Svy_est(f, i))) / (df * square(Svy_se_use(f, i))))) + sd_offset * log(Svy_se_use(f, i));
-            }
-        }
-        else if (Svy_errtype(f) == -1) // normal
-        {
-          for (i = 1; i <= Svy_N_fleet(f); i++)
+        for (i = 1; i <= Svy_N_fleet(f); i++)
+          if (Svy_use(f, i) > 0)
           {
-            if (Svy_use(f, i) > 0)
-            {
-              surv_like(f) += 0.5 * square((Svy_obs(f, i) - Svy_est(f, i)) / Svy_se_use(f, i)) + sd_offset * log(Svy_se_use(f, i));
+            if (Svy_errtype(f) == 0) {  // lognormal
+              Svy_like_I(f,i) = 0.5 * square((Svy_obs_log(f, i) - Svy_est(f, i)) / Svy_se_use(f, i))
+                              + sd_offset * log(Svy_se_use(f, i));
+             }
+            else if (Svy_errtype(f) == 1) {  // lognormal with bias adjustment
+              Svy_like_I(f,i) = 0.5 * square((Svy_obs_log(f, i) - Svy_est(f, i) + 0.5 * Svy_se_use(f, i) * Svy_se_use(f, i) ) / Svy_se_use(f, i))
+                              + sd_offset * log(Svy_se_use(f, i));
+              }
+            else if (Svy_errtype(f) >= 1000) {  // T-dist
+              dvariable df = Svy_errtype(f) - 1000.;
+              Svy_like_I(f,i) = ((df + 1.) / 2.) * log((1. + square((Svy_obs_log(f, i) - Svy_est(f, i))) / (df * square(Svy_se_use(f, i)))))
+                              + sd_offset * log(Svy_se_use(f, i));
+              }
+            else if (Svy_errtype(f) == -1) {  // normal
+              Svy_like_I(f,i) =  0.5 * square((Svy_obs(f, i) - Svy_est(f, i)) / Svy_se_use(f, i))
+                              + sd_offset * log(Svy_se_use(f, i));
             }
           }
```

Comment:
Maybe this is caught earlier but what if an invalid Svy_errtype is entered? Say a -5? It looks like it'll just skip the calculations but really should error out or warn?